File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {

        int l = s.size() ; 
        int hl = l/2 ; 


        string set = "aeiouAEIOU";

        int left = 0 , right = 0 ; 
        for(int i=0; i <=l ; i++){
            if( i <hl && set.find(s[i]) != string::npos) { 
                left++ ; 
            }else if( i>=hl && set.find(s[i]) != string::npos ){
                right++; 
            }
        }
        return left ==right ; 
    }
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        
        
        var vowels = Set.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U');
        int left = 0 , right =0 ; 
        for(int i=0 , j = s.length() -1 ;  i< j ; i++ , j--){
            left += vowels.contains(s.charAt(i)) ? 1 : 0 ; 
            right += vowels.contains(s.charAt(j)) ? 1 : 0 ; 
        }
        return left == right ; 
    }
}